salt/pillar/__init__.py

```diff
@@ -302,6 +298,119 @@ def __init__(self, opts, grains, minion_id, saltenv, ext=None, functions=None,
             else:
                 log.error('Pillar data must be a dictionary')
 
+    @property
```

Comment:
``self.pillar`` is marked as a ``@property`` - that means that even though the code references ``self.pillar`` as if it's simply referencing data, the data is generated and returned by a function.